src/plugins/dashboard/public/application/actions/clone_panel_action.test.tsx

```diff
@@ -113,56 +135,156 @@ test('Clone adds a new embeddable', async () => {
+  expect(loadedPanel.type).toEqual(byRefOrValEmbeddable.type);
 });
 
-test('Clones an embeddable without a saved object ID', async () => {
-  const dashboard = embeddable.getRoot() as IContainer;
-  const panel = dashboard.getInput().panels[embeddable.id] as DashboardPanelState;
+test('Clones a RefOrVal embeddable by value', async () => {
+  const dashboard = byRefOrValEmbeddable.getRoot() as IContainer;
+  const panel = dashboard.getInput().panels[byRefOrValEmbeddable.id] as DashboardPanelState;
   const action = new ClonePanelAction(coreStart);
   // @ts-ignore
-  const newPanel = await action.cloneEmbeddable(panel, embeddable.type);
-  expect(newPanel.type).toEqual(embeddable.type);
+  const newPanel = await action.cloneEmbeddable(panel, byRefOrValEmbeddable);
+  expect(coreStart.savedObjects.client.get).toHaveBeenCalledTimes(0);
+  expect(coreStart.savedObjects.client.find).toHaveBeenCalledTimes(0);
+  expect(coreStart.savedObjects.client.create).toHaveBeenCalledTimes(0);
+  expect(newPanel.type).toEqual(byRefOrValEmbeddable.type);
 });
 
-test('Clones an embeddable with a saved object ID', async () => {
-  const dashboard = embeddable.getRoot() as IContainer;
-  const panel = dashboard.getInput().panels[embeddable.id] as DashboardPanelState;
+test('Clones a non-RefOrVal embeddable by value if the panel does not have a savedObjectId', async () => {
+  const dashboard = genericEmbeddable.getRoot() as IContainer;
+  const panel = dashboard.getInput().panels[genericEmbeddable.id] as DashboardPanelState;
+  const action = new ClonePanelAction(coreStart);
+  // @ts-ignore
+  const newPanelWithoutId = await action.cloneEmbeddable(panel, genericEmbeddable);
+  expect(coreStart.savedObjects.client.get).toHaveBeenCalledTimes(0);
+  expect(coreStart.savedObjects.client.find).toHaveBeenCalledTimes(0);
+  expect(coreStart.savedObjects.client.create).toHaveBeenCalledTimes(0);
+  expect(newPanelWithoutId.type).toEqual(genericEmbeddable.type);
+});
+
+test('Clones a non-RefOrVal embeddable by reference if the panel has a savedObjectId', async () => {
+  const dashboard = genericEmbeddable.getRoot() as IContainer;
+  const panel = dashboard.getInput().panels[genericEmbeddable.id] as DashboardPanelState;
   panel.explicitInput.savedObjectId = 'holySavedObjectBatman';
   const action = new ClonePanelAction(coreStart);
   // @ts-ignore
-  const newPanel = await action.cloneEmbeddable(panel, embeddable.type);
+  const newPanel = await action.cloneEmbeddable(panel, genericEmbeddable);
   expect(coreStart.savedObjects.client.get).toHaveBeenCalledTimes(1);
   expect(coreStart.savedObjects.client.find).toHaveBeenCalledTimes(1);
   expect(coreStart.savedObjects.client.create).toHaveBeenCalledTimes(1);
-  expect(newPanel.type).toEqual(embeddable.type);
+  expect(newPanel.type).toEqual(genericEmbeddable.type);
 });
 
-test('Gets a unique title ', async () => {
+test('Gets a unique from the saved objects library', async () => {
```

Comment:
Nit: should this be `Gets a unique **title** from the saved objects library'?

Reply:
:grimacing: Yup, you're totally right. Will fix this.